File: Day-33/ColorPicker OBJ-C/ColorPicker OBJ-C/ViewController.m
//
//  ViewController.m
//  ColorPicker OBJ-C
//
//  Created by Bronson Dupaix on 3/16/16.
//  Copyright © 2016 Bronson Dupaix. All rights reserved.
//

#import "ViewController.h"
#import "ISColorWheel.h"
#import "Color.h"

@interface ViewController () <ISColorWheelDelegate>

@property (strong, nonatomic) ISColorWheel *colorWheel;

@property (strong, nonatomic) NSMutableArray *colorArray;

// @property (strong, nonatomic) Color* color;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.colorArray = [[NSMutableArray alloc] init];
    
    CGSize size = self.view.bounds.size;
    
    
    CGSize wheelSize = CGSizeMake(size.width * .65, size.width * .65);
    
    _colorWheel = [[ISColorWheel alloc] initWithFrame: CGRectMake(size.width / 2 - wheelSize.width / 2,
                                                                 size.height * .15,
                                                                 wheelSize.width,
                                                                 wheelSize.height)];
    
    
    _colorWheel.delegate = self;
    _colorWheel.continuous = true;
    [self.view addSubview:_colorWheel];
    
   
}

- (void)colorWheelDidChangeColor:(ISColorWheel *)colorWheel
{
    
    self.view.backgroundColor = [colorWheel currentColor];
    UIColor *uicolor = [colorWheel currentColor] ;
    CGColorRef color = [uicolor CGColor];
    
    int numComponents = CGColorGetNumberOfComponents(color);
    
    if (numComponents == 3)
    {
        const CGFloat *components = CGColorGetComponents(color);
        CGFloat red = components[0];
        CGFloat green = components[1];
        CGFloat blue = components[2];
        NSLog(@"red: %.2f" ,red);
        NSLog(@"blue: %.2f" ,blue);
        NSLog(@"green: %.2f" ,green); 
        
        [self.colorArray addObject:[UIColor colorWithRed:red green:green blue:blue alpha:1.0f]];
        
        // self.color.red = red
        
    }
}

-(UITableViewCell *) tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    
    
     UIColor *color = [self.colorArray objectAtIndex:indexPath.row];
    
    UITableViewCell * cell = [tableView dequeueReusableCellWithIdentifier:@"ColorCell"];
    
    cell.textLabel.text = @"Hello"; 
    
    // cell.backgroundColor = [UIColor _color];
    
    return cell;
}


-(NSInteger) tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section{
    return 1;
}

@end

File: MovieDBApp2/MovieDBApp2/ViewController.swift
//
//  ViewController.swift
//  MovieDBApp2
//
//  Created by Bronson Dupaix on 2/24/16.
//  Copyright © 2016 Bronson Dupaix. All rights reserved.
//

import UIKit


protocol MovieProtocol{
    
    func passMovie(movieArray: [Movie]) 
    
}

class ViewController: UIViewController, MovieProtocol, UITableViewDataSource, UITableViewDelegate {
    
    var moviesArray = [Movie]()
    
    var api = MovieAPI()

    @IBOutlet weak var movieNameTableVIew: UITableView!
    
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        
        api.delegate = self 
        
        api.fetchMovies()
      
    } // end of view did load 
    
    func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
        
        let movie = self.moviesArray[indexPath.row]
        
        let cell = UITableViewCell() 
        
        cell.textLabel?.text = movie.title
        
        return cell
    }
    
    func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return moviesArray.count
    }
    
    func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
        
    }
    
    func passMovie(movieArray: [Movie]) {
        
        self.moviesArray = movieArray
        
        
        
        dispatch_async(dispatch_get_main_queue(), {
            
            self.movieNameTableVIew.reloadData()
            
        })

    }

   

} // end of view

